### PR TITLE
Change to upstream tzdata

### PR DESCRIPTION
**Pull request by mbargull**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

This changes this repository to build the upstream Time Zone Database as the package named `tzdata` on conda-forge.
Up until now, this recipe built the Python package `tzdata`, which is now living at https://github.com/conda-forge/python-tzdata-feedstock and will be available under the `python-tzdata` name on the `conda-forge` channel.
Following up on: https://github.com/conda-forge/python-feedstock/issues/392 and https://github.com/conda-forge/staged-recipes/pull/12828

### CODE REVIEW

File: recipe/meta.yaml
Comment (mbargull) [inline recipe/meta.yaml:36]: These files are included in the Python package. Which means this initial build will carry the same set of files (albeit under a different path) as https://pypi.org/project/tzdata/2020.1/ .

I don't know if their inclusion in the Python package was intentional or not. (And I have no experience in dealing with tzdata, so can't tell when they are needed.)
I'll open an issue for that at their issue tracker.

File: recipe/meta.yaml
Comment (mbargull) [inline recipe/meta.yaml:1]: Although `conda` also uses Python-like versioning, `tzdata` uses consistent versions for which the non-numeric suffix is always lexicographical increasing.
Meaning: It's fine for us to use the upstream version scheme.

File: recipe/meta.yaml
Comment (mbargull) [inline recipe/meta.yaml:25]: This uses a compiler because it builds its own tools. We add the `ignore_run_exports` above to get a `noarch: generic` package without any other dependencies.

File: recipe/meta.yaml
Comment (mbargull) [inline recipe/meta.yaml:21]: This is Linux-specific, of course but we can't use `skip` since the linter complains.

Comment (mbargull) [reply recipe/meta.yaml:21]: I'll add a comment in the recipe for this.

File: recipe/build.sh
Comment (pganssle) [inline recipe/build.sh:5]: I would use `-b fat` (or whatever the equivalent is) for now. `pytz` and `dateutil` will have problems with the slim versions, unfortunately.

File: recipe/0001-Add-TABDATA-to-install_data.patch
Comment (pganssle) [inline recipe/0001-Add-TABDATA-to-install_data.patch:7]: Why do you need to patch this?

File: recipe/meta.yaml
Comment (pganssle) [reply recipe/meta.yaml:39]: Which of these do you think shouldn't be installed?

`tzdata.zi` definitely should be, the tab files and leap seconds files were included intentionally as they are part of the installed distribution (we take whatever the upstream project installs, see [this](https://github.com/python/tzdata/blob/842c5384d0a02c30de513b33277369d488f374c2/update.py#L86-L99)).

File: recipe/build.sh
Comment (mbargull) [reply recipe/build.sh:5]: Ah, thanks. I assume you intend to change the Python package to `-b fat` then too?

File: recipe/0001-Add-TABDATA-to-install_data.patch
Comment (mbargull) [reply recipe/0001-Add-TABDATA-to-install_data.patch:7]: Just to avoid the manual copying of those extra files ;).
But if we don't need them, we can remove that whole commit. I just don't know if it makes sense to include them or not.
(ref: https://github.com/python/tzdata/issues/28)

File: recipe/build.sh
Comment (pganssle) [reply recipe/build.sh:5]: No, the Python package is not used by anything except `zoneinfo` at the moment, whereas `dateutil` will automatically use the system time zone information, and `pytz` has an option to use the system time zone data. Anyone who starts using the Python `tzdata` package should support slim files.

File: recipe/meta.yaml
Comment (mbargull) [reply recipe/meta.yaml:39]: Oh, maybe it makes sense to include them all. Unfortunately I have no in-depth knowledge of this at all.
Hence I included all files you have in https://github.com/python/tzdata in this one as well.

Comment (mbargull) [reply recipe/meta.yaml:39]: Thanks for reviewing this an confirming that it was indeed intentional!

File: recipe/0001-Add-TABDATA-to-install_data.patch
Comment (pganssle) [reply recipe/0001-Add-TABDATA-to-install_data.patch:7]: I'm not sure why you have to patch anything to get those files. Upstream the way I unpack this is [here](https://github.com/python/tzdata/blob/842c5384d0a02c30de513b33277369d488f374c2/update.py#L86-L99). I do:

`make DESTDIR={temporary_directory} POSIXRULES= ZFLAGS='-b slim' install`

(You would use `-b fat`) and then move everything that's in `temporary_directory/usr/share/zoneinfo` over to the location I'm expecting. As far as I know no additional copying is necessary.

**Comment by ocefpaf** [comment]
@mbargull thanks so much for doing this! I was planning to hack at it over the weekend but could not even get close to reading the build intructions.

I would like to say I approve and this is ready to go but you have the much better expertise of @pganssle here so I'll defer to him to tell us when this is OK to merge.

File: recipe/0001-Add-TABDATA-to-install_data.patch
Comment (mbargull) [reply recipe/0001-Add-TABDATA-to-install_data.patch]: I just wanted a Make target that installed everything in `TZDIR`. But I'm also more than fine with the way you did it and added a commit to do just that!
The tests in this recipe will also catch when anything unexpected would get installed, so all good either way.

**Comment by mbargull** [comment]
Thanks @ocefpaf. And yes, I'm also very happy that @pganssle is helping out here, much appreciated!

**Review by ocefpaf — APPROVED** [review]

**Comment by ocefpaf** [comment]
@pganssle any last comments or is this OK to be merged?

**Comment by mbargull** [comment]
This is likely good to go now -- anything else we should change, @pganssle? Would you like to help maintaining this recipe, too? I took you off the maintainers list just to not enforce an additional recipe onto you -- but of course we'd be happy to have you!

Just to recap what's different from https://github.com/conda-forge/python-tzdata:
- This is not a `noarch: python` package anymore, but just a `noarch: generic` with all files under `share/zoneinfo`.
- This doesn't carry the `zones` files from the Python package.
- The actual data files are the same as in the Python package, but not built with `-b slim`.
- We can use this as a dependency for the `python` package and point its `TZPATH` to the directory from this package then.

File: recipe/meta.yaml
Comment (jakirkham) [inline recipe/meta.yaml:41]: Was this intended?

Comment (ocefpaf) [reply recipe/meta.yaml:41]: Yep: https://github.com/conda-forge/tzdata-feedstock/pull/1#issuecomment-704409215

Comment (ocefpaf) [reply recipe/meta.yaml:41]: BTW, I already "forced" @pganssle in the original tzdata, now python-tzdata, @mbargull is doing the polite thing by asking him first since this package was changed in-place.

Comment (mbargull) [reply recipe/meta.yaml:41]: See https://github.com/conda-forge/tzdata-feedstock/pull/1#issuecomment-704409215:
> ..., @pganssle? Would you like to help maintaining this recipe, too? I took you off the maintainers list just to not enforce an additional recipe onto you -- but of course we'd be happy to have you!

Since I moved the recipe that he "signed up for" to https://github.com/conda-forge/python-feedstock, this one is essentially a "new" one. I'd be very happy to have @pganssle as a maintainer for this one too, but did not want to impose any role without asking!

Comment (mbargull) [reply recipe/meta.yaml:41]: > BTW, I already "forced" @pganssle ...

:laughing:. I only now read https://github.com/conda-forge/staged-recipes/pull/11801:
You did write `@pganssle do you want to be a maintainer here?` there -- you did the polite thing too :wink:!
BTW, it's good to see that [@pganssle's comment there](https://github.com/conda-forge/staged-recipes/pull/11801#issuecomment-637569942) also reflects what we're doing here now, which is a nice confirmation!

Comment (jakirkham) [reply recipe/meta.yaml:41]: Oh ok cool. Sorry I had not read everything and thought this might be a typo. Feel free to mark resolved.

**Comment by ocefpaf** [comment]
Let's get this in. We can always add/remove stuff later. Thanks @mbargull for fixing this and @pganssle for the valuable insight!

**Comment by pganssle** [comment]
> This is likely good to go now -- anything else we should change, @pganssle? Would you like to help maintaining this recipe, too? I took you off the maintainers list just to not enforce an additional recipe onto you -- but of course we'd be happy to have you!

Hey, sorry, this seemed low-priority since it was merged before I got back to it, but yeah I'm fine helping to maintain this feedstock.

Thanks for putting this together, I think it looks great!

(Now for the first test — `tzdata` 2020b was just released!)

**Pull request merged**
